executor/src/witgen/mod.rs

```diff
@@ -225,7 +233,7 @@ impl<'a, 'b, T: FieldElement> WitnessGenerator<'a, 'b, T> {
             mut machines,
             base_parts,
         } = if self.stage == 0 {
-            MachineExtractor::new(&fixed).split_out_machines(retained_identities, self.stage)
+            MachineExtractor::new(&fixed).split_out_machines(retained_identities)
```

Comment:
So we don't pass the stage because we already only pass the identities up to this stage

Reply:
Yes, and also the machine extractor can use functions like `FixedData::current_stage_witnesses`.